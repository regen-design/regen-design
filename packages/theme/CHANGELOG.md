# @regen-design/theme

## 1.0.0

### Major Changes

- publish
