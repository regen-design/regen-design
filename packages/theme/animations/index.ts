import { keyframes } from 'styled-components'

export const ButtonWaveSpreadAnimationKeyframes = (color: string) => keyframes`
    0% {
        opacity: 0.6;
        box-shadow: 0 0 1px 0 ${color};
    }
    100% {
        opacity: 0;
        box-shadow: 0 0 1px 5px ${color};
    }
`
