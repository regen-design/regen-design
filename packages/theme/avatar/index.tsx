import { NAME_SPACE } from '@regen-design/constant'
import { AvatarProps } from '@regen-design/types'
import styled from 'styled-components'
import { convertTheme } from '../tools'
const prefix = `${NAME_SPACE}-avatar`
export const StyledAvatarPrefixClass = prefix
export const StyledAvatar = styled.div<AvatarProps>`
  display: inline-flex;
  position: relative;
  overflow: hidden;
  text-align: center;
  border: ${props => {
    const _theme = convertTheme(props.theme)
    return `1px solid ${_theme.colors[props.borderColor || 'transparent']}`
  }};
  width: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.components.avatar.size[props.size || 'default']
  }};
  height: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.components.avatar.size[props.size || 'default']
  }};
  color: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.colors[props.color || 'white']
  }};
  font-size: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.components.avatar.fontSize
  }};
  border-radius: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.rounded) return '50%'
    return _theme.borderRadius
  }};
  background-color: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.colors[props.backgroundColor || 'placeholder']
  }};
  img {
    width: 100%;
    height: 100%;
  }
  .${prefix}__text {
    position: absolute;
    line-height: 1;
    white-space: nowrap;
    display: inline-block;
    left: 50%;
    top: 50%;
  }
`
