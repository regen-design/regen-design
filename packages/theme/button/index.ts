import styled from 'styled-components'
import { convertTheme } from '../tools'
import { ButtonProps } from '@regen-design/types'
import { NAME_SPACE } from '@regen-design/constant'
import { lighten } from 'polished'
const prefix = `${NAME_SPACE}-button`

export const StyledButtonPrefixClass = prefix

export const StyledButton = styled.button<ButtonProps>`
  background-color: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.text) {
      return 'transparent'
    }
    if (props.type === 'default') {
      return _theme.theme === 'light' ? _theme.colors.light : _theme.colors.dark
    }
    if (props.dashed) {
      return 'transparent'
    }
    return _theme.colors[props.type]
  }};
  border-radius: ${props => {
    return convertTheme(props.theme).borderRadius + 'px'
  }};
  color: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.text) {
      return _theme.colors[props.type]
    }
    if (props.type === 'default') {
      return _theme.theme === 'light' ? _theme.colors.dark : _theme.colors.light
    }
    if (props.dashed) {
      return `${_theme.colors[props.type]}`
    }
    return _theme.theme === 'light' ? _theme.colors.light : _theme.colors.dark
  }};
  &:hover {
    color: ${props => {
      const _theme = convertTheme(props.theme)
      if (props.type === 'default') {
        return lighten(0.025, _theme.colors.primary)
      }
      if (props.text) {
        return lighten(0.05, _theme.colors?.[props.type] || '#000000')
      }
    }};
  }
  &:active {
    color: ${props => {
      const _theme = convertTheme(props.theme)
      if (props.type === 'default') {
        return lighten(0.15, _theme.colors.primary)
      }
      if (props.text) {
        return lighten(0.25, _theme.colors?.[props.type] || '#000000')
      }
      if (props.dashed) {
        return `${_theme.colors[props.type]}`
      }
      return _theme.theme === 'light' ? _theme.colors.light : _theme.colors.dark
    }};
  }
  transition: ${props => {
    const duration = `${convertTheme(props.theme).animations.duration}ms`
    return `all ${props.text ? '50ms' : duration} ease-in`
  }};
  opacity: ${props => {
    if (props.disabled) {
      return 0.5
    }
  }};
  display: inline-flex;
  flex-wrap: nowrap;
  white-space: nowrap;
  outline: none;
  position: relative;
  z-index: auto;
  border: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.text) {
      return 'transparent'
    }
    if (props.type === 'default') {
      const color = _theme.theme === 'light' ? _theme.borderColor : _theme.borderDarkColor
      return `1px ${props.dashed ? 'dashed' : 'solid'} ${color}`
    }
    if (props.dashed) {
      return `1px dashed ${_theme.colors[props.type]}`
    }
    return '1px solid transparent'
  }};
  padding: ${props => {
    if (props.text) {
      return '0 0'
    }
    switch (props.size) {
      case 'tiny':
        return '0 8px'
      case 'small':
        return '0 12px'
      case 'large':
        return '0 16px'
      default:
        return '0 12px'
    }
  }};
  height: ${props => {
    if (props.text) {
      return 'initial'
    }
    switch (props.size) {
      case 'tiny':
        return '24px'
      case 'small':
        return '32px'
      case 'large':
        return '40px'
      default:
        return '36px'
    }
  }};
  line-height: 1;
  font-family: inherit;
  font-size: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.fontSizes[props.size]
  }};
  width: initial;
  align-items: center;
  justify-content: center;
  user-select: none;
  -webkit-user-select: none;
  text-align: center;
  cursor: ${props => {
    if (props.disabled) {
      return 'not-allowed'
    }
    return 'pointer'
  }};
  text-decoration: none;
  .${prefix}__content {
    display: flex;
    align-items: center;
    flex-wrap: nowrap;
    min-width: 0;
  }
  .${prefix}__icon {
    margin-right: 8px;
  }
`
