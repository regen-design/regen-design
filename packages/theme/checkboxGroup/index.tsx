import { NAME_SPACE } from '@regen-design/constant'
import { CheckboxGroupProps } from '@regen-design/types'
import styled from 'styled-components'
const prefix = `${NAME_SPACE}-checkboxGroup`
export const StyledCheckboxGroupPrefixClass = prefix
export const StyledCheckboxGroup = styled.div<CheckboxGroupProps>``
