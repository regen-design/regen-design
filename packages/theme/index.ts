import { ThemeConfig } from '@regen-design/types'
import { useTheme as useThemeFn } from 'styled-components'
export const useTheme = useThemeFn as () => ThemeConfig
export * from './animations'
export * from './button'
export * from './card'
export * from './checkbox'
export * from './checkboxGroup'
export * from './codeBlock'
export * from './grid'
export * from './gridItem'
export * from './icon'
export * from './input'
export * from './radio'
export * from './radioGroup'
export * from './space'
export * from './variable'
export * from './wave'
