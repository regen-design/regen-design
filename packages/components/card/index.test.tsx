import '@testing-library/jest-dom/vitest'
import { describe, expect, test } from 'vitest'
import { render, screen } from '@testing-library/react'
import { Card } from '.'
import { StyledCardPrefixClass as prefixClass } from '@regen-design/theme'
describe('Card', () => {
  test('should work with `title` prop', () => {
    const element = render(<Card title="title" />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('card-header')).toHaveTextContent('title')
  })
  test('should work with `extra` prop', () => {
    const element = render(<Card extra="extra" />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('card-header')).toHaveTextContent('extra')
  })
  test('should work with `footer` prop', () => {
    const element = render(<Card footer="footer" />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('card-footer')).toHaveTextContent('footer')
  })
  test('should work with `size` prop', () => {
    const element = render(<Card size="small" />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('card')).toHaveClass(`${prefixClass}--small`)
  })
  test('should work with `className` prop', () => {
    const element = render(<Card className="test" />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('card')).toHaveClass('test')
  })
  test('should work with `style` prop', () => {
    const element = render(<Card style={{ color: 'red' }} />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('card')).toHaveStyle({ color: 'red' })
  })
  test('should work with `footerStyle` prop', () => {
    const element = render(<Card footerStyle={{ color: 'red' }} footer="footer" />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('card-footer')).toHaveStyle({ color: 'red' })
  })
  test('should work with `footerClassName` prop', () => {
    const element = render(<Card footerClassName="test" footer="footer" />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('card-footer')).toHaveClass('test')
  })
  test('should work with `children` prop', () => {
    const element = render(<Card>children</Card>)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('card-body')).toHaveTextContent('children')
  })
})
