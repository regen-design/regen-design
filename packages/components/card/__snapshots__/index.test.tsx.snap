// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should work with \`children\` prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLQSwg inAgOO regen-card regen-card--default"
        role="card"
      >
        <div
          class="regen-card-header"
          role="card-header"
        >
          <div
            class="regen-card-header-title"
          />
          <div
            class="regen-card-header-extra"
          />
        </div>
        <div
          class="regen-card-body"
          role="card-body"
        >
          children
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLQSwg inAgOO regen-card regen-card--default"
      role="card"
    >
      <div
        class="regen-card-header"
        role="card-header"
      >
        <div
          class="regen-card-header-title"
        />
        <div
          class="regen-card-header-extra"
        />
      </div>
      <div
        class="regen-card-body"
        role="card-body"
      >
        children
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card > should work with \`className\` prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLQSwg inAgOO regen-card test regen-card--default"
        role="card"
      >
        <div
          class="regen-card-header"
          role="card-header"
        >
          <div
            class="regen-card-header-title"
          />
          <div
            class="regen-card-header-extra"
          />
        </div>
        <div
          class="regen-card-body"
          role="card-body"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLQSwg inAgOO regen-card test regen-card--default"
      role="card"
    >
      <div
        class="regen-card-header"
        role="card-header"
      >
        <div
          class="regen-card-header-title"
        />
        <div
          class="regen-card-header-extra"
        />
      </div>
      <div
        class="regen-card-body"
        role="card-body"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card > should work with \`extra\` prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLQSwg inAgOO regen-card regen-card--default"
        role="card"
      >
        <div
          class="regen-card-header"
          role="card-header"
        >
          <div
            class="regen-card-header-title"
          />
          <div
            class="regen-card-header-extra"
          >
            extra
          </div>
        </div>
        <div
          class="regen-card-body"
          role="card-body"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLQSwg inAgOO regen-card regen-card--default"
      role="card"
    >
      <div
        class="regen-card-header"
        role="card-header"
      >
        <div
          class="regen-card-header-title"
        />
        <div
          class="regen-card-header-extra"
        >
          extra
        </div>
      </div>
      <div
        class="regen-card-body"
        role="card-body"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card > should work with \`footer\` prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLQSwg inAgOO regen-card regen-card--default"
        role="card"
      >
        <div
          class="regen-card-header"
          role="card-header"
        >
          <div
            class="regen-card-header-title"
          />
          <div
            class="regen-card-header-extra"
          />
        </div>
        <div
          class="regen-card-body"
          role="card-body"
        />
        <div
          class="regen-card-footer "
          role="card-footer"
        >
          footer
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLQSwg inAgOO regen-card regen-card--default"
      role="card"
    >
      <div
        class="regen-card-header"
        role="card-header"
      >
        <div
          class="regen-card-header-title"
        />
        <div
          class="regen-card-header-extra"
        />
      </div>
      <div
        class="regen-card-body"
        role="card-body"
      />
      <div
        class="regen-card-footer "
        role="card-footer"
      >
        footer
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card > should work with \`footerClassName\` prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLQSwg inAgOO regen-card regen-card--default"
        role="card"
      >
        <div
          class="regen-card-header"
          role="card-header"
        >
          <div
            class="regen-card-header-title"
          />
          <div
            class="regen-card-header-extra"
          />
        </div>
        <div
          class="regen-card-body"
          role="card-body"
        />
        <div
          class="regen-card-footer test"
          role="card-footer"
        >
          footer
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLQSwg inAgOO regen-card regen-card--default"
      role="card"
    >
      <div
        class="regen-card-header"
        role="card-header"
      >
        <div
          class="regen-card-header-title"
        />
        <div
          class="regen-card-header-extra"
        />
      </div>
      <div
        class="regen-card-body"
        role="card-body"
      />
      <div
        class="regen-card-footer test"
        role="card-footer"
      >
        footer
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card > should work with \`footerStyle\` prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLQSwg inAgOO regen-card regen-card--default"
        role="card"
      >
        <div
          class="regen-card-header"
          role="card-header"
        >
          <div
            class="regen-card-header-title"
          />
          <div
            class="regen-card-header-extra"
          />
        </div>
        <div
          class="regen-card-body"
          role="card-body"
        />
        <div
          class="regen-card-footer "
          role="card-footer"
          style="color: red;"
        >
          footer
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLQSwg inAgOO regen-card regen-card--default"
      role="card"
    >
      <div
        class="regen-card-header"
        role="card-header"
      >
        <div
          class="regen-card-header-title"
        />
        <div
          class="regen-card-header-extra"
        />
      </div>
      <div
        class="regen-card-body"
        role="card-body"
      />
      <div
        class="regen-card-footer "
        role="card-footer"
        style="color: red;"
      >
        footer
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card > should work with \`size\` prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLQSwg hAcxZE regen-card regen-card--small"
        role="card"
      >
        <div
          class="regen-card-header"
          role="card-header"
        >
          <div
            class="regen-card-header-title"
          />
          <div
            class="regen-card-header-extra"
          />
        </div>
        <div
          class="regen-card-body"
          role="card-body"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLQSwg hAcxZE regen-card regen-card--small"
      role="card"
    >
      <div
        class="regen-card-header"
        role="card-header"
      >
        <div
          class="regen-card-header-title"
        />
        <div
          class="regen-card-header-extra"
        />
      </div>
      <div
        class="regen-card-body"
        role="card-body"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card > should work with \`style\` prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLQSwg inAgOO regen-card regen-card--default"
        role="card"
        style="color: red;"
      >
        <div
          class="regen-card-header"
          role="card-header"
        >
          <div
            class="regen-card-header-title"
          />
          <div
            class="regen-card-header-extra"
          />
        </div>
        <div
          class="regen-card-body"
          role="card-body"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLQSwg inAgOO regen-card regen-card--default"
      role="card"
      style="color: red;"
    >
      <div
        class="regen-card-header"
        role="card-header"
      >
        <div
          class="regen-card-header-title"
        />
        <div
          class="regen-card-header-extra"
        />
      </div>
      <div
        class="regen-card-body"
        role="card-body"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card > should work with \`title\` prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLQSwg inAgOO regen-card regen-card--default"
        role="card"
      >
        <div
          class="regen-card-header"
          role="card-header"
        >
          <div
            class="regen-card-header-title"
          >
            title
          </div>
          <div
            class="regen-card-header-extra"
          />
        </div>
        <div
          class="regen-card-body"
          role="card-body"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLQSwg inAgOO regen-card regen-card--default"
      role="card"
    >
      <div
        class="regen-card-header"
        role="card-header"
      >
        <div
          class="regen-card-header-title"
        >
          title
        </div>
        <div
          class="regen-card-header-extra"
        />
      </div>
      <div
        class="regen-card-body"
        role="card-body"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
